server/routerlicious/packages/lambdas/src/scribe/lambda.ts

```diff
@@ -345,12 +374,30 @@ export class ScribeLambda implements IPartitionLambda {
             }
         }
 
-        const checkpoint = this.generateCheckpoint(message.offset);
-        this.checkpointCore(
-            checkpoint,
-            message,
-            this.clearCache);
-        this.lastOffset = message.offset;
+        // Create the checkpoint
+        const checkpoint = this.generateScribeCheckpoint(message.offset);
+        this.checkpointInfo.rawMessagesSinceCheckpoint++;
+        this.updateCheckpointMessages(message);
+        const checkpointReason = this.getCheckpointReason();
```

Comment:
Nit: Can we avoid doing this computation for the noActiveClients client since we already have the checkpointReason?

Reply:
Updated so we only call getCheckpointReason() after we check if there are noActiveClients